### ISSUE TITLE
How to easily save and recall optimizer and scheduler checkpoints

### ISSUE THREAD

**Issue by mhy9989** [microsoft/DeepSpeed#4834]
I tried to save the checkpoint without using deepspeed.DeepSpeedEngine.save_checkpoint(). The checkpoint is saved after the deepspeed.initialize().
The code is:
```
checkpoint = {  
    'epoch':  epoch + 1,  
     'optimizer': optimizer.state_dict(),  
     'state_dict': model.module.state_dict(),  
     'scheduler': scheduler.state_dict()  
}  
torch.save(checkpoint, 'checkpoint.pth') 
```
Restart the program.  
When I want to load the checkpoint, I use the following code (after the deepspeed.initialize()): 
```
checkpoint = torch.load(checkpoint.pth)
model.module.load_state_dict(checkpoint['state_dict'])
optimizer.load_state_dict(checkpoint['optimizer'])
scheduler.load_state_dict(checkpoint['scheduler'])
```
Then I found that the training speed was several times slower.  
I do not know why it has to be like this.  

If I load the checkpoint before the deepspeed.initialize(), the model can load the checkpoint normally, but the optimizer and scheduler will have a state_dict device (cuda) error after deepspeed.initialize().  
Even if I move the state_dict of the optimizer and scheduler to device (cuda) before deepspeed.initialize(), the training speed will still slow down afterwards.  

So, I want to ask how to solve this problem?

**Comment by mhy9989** [comment]
Moreover, I also found that loading the optimizer state_dict will occupy more GPU memory than training directly. How to solve this?

**Comment by ra1995** [comment]
@mhy9989 Hi, could you kindly let me know how you solved this issue?
